reservations/schema.py

```diff
@@ -58,18 +69,34 @@ class ReservationInput(graphene.InputObjectType):
 class CreateReservation(graphene.Mutation):
     class Arguments:
         reservation = ReservationInput()
+        switch = BerthSwitchInput()
 
     ok = graphene.Boolean()
     reservation = graphene.Field(ReservationType)
 
     def mutate(self, info, **kwargs):
         reservation_data = kwargs.pop("reservation")
+        switch_data = kwargs.pop("switch")
         try:
             obj_id = reservation_data.pop("boat_type")
         except KeyError:
             obj_id = None
         if obj_id:
             reservation_data["boat_type"] = BoatType.objects.get(id=obj_id)
+
+        try:
```

Comment:
getting a bit hard to read with all these try blocks. if you have time/energy, you could refactor this maybe, or I will refactor it later at some point.